server/src/main/java/io/druid/segment/realtime/firehose/PredicateFirehose.java

```diff
@@ -55,6 +60,11 @@ public boolean hasMore()
         savedInputRow = row;
         return true;
       }
+      // Do not silently discard the rows
+      if (ignored % ignoredThreshold == 0) {
```

Comment:
this means if only the first 5000 events are not properly formatted you won't see the log. i will use an exponential backoff strategy.